packages/kbn-optimizer/src/cli.ts

```diff
@@ -119,7 +127,13 @@ run(
       update$ = update$.pipe(reportOptimizerStats(reporter, config, log));
     }
 
+    update$ = update$.pipe(apmOptimizerStats(config));
+
     await update$.pipe(logOptimizerState(log, config)).toPromise();
+    await apm.flush();
+
+    // TODO: It appears the cb is called BEFORE the API request in flush
+    await new Promise((resolve) => setTimeout(resolve, 1000));
```

Comment:
@vigneshshanmugam, this is the issue I mentioned previously. I haven't dug into it too deeply, but it appears the callback to flush is called before the HTTP request is made. Is this something you could look into? A 1-second wait is obviously something we don't want to merge with.

Reply:
Sure, I will have a look.

Reply:
Were you able to reproduce?

Reply:
Was busy with other stuffs, Apologies for the delay. Will get back to it soon. Is it blocking the current PR?

Reply:
Hey, yeah we will need to resolve that before moving forward with this. We're currently working to have APM collection for local development, CI and on Cloud.

Reply:
Sure, Had a quick look and it seems like since there is no active request to the APM server, the callback is getting immediately invoked as a result. I will debug more on how we can tune it to report events from every CLI run.

Reply:
Btw, I did try a few runs and all of the events were reported properly to the server even without the setTimeout hack as the underlying transport layer guarantees the transaction was written to the stream when `trans.end()` is called and flush guarantees the stream is closed properly. 

Do we need to do anything else here?